src/components/timeline/TrackVisualProgress.js

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import explicitConnect from '../../utils/connect';
+import { formatPercent } from '../../utils/format-numbers';
+import {
+  getCommittedRange,
+  getVisualProgressSelectors,
+} from '../../selectors/profile';
+import { updatePreviewSelection } from '../../actions/profile-view';
+import {
+  TRACK_VISUAL_PROGRESS_HEIGHT,
+  TRACK_COUNTER_LINE_WIDTH,
+} from '../../app-logic/constants';
+import { TrackCounterGraph, GRAPH_TYPE } from './TrackCounterGraph';
+import { BLUE_50, BLUE_60 } from 'photon-colors';
+
+import type { CounterIndex, ThreadIndex, Counter } from '../../types/profile';
+import type { Milliseconds } from '../../types/units';
+import type { ConnectedProps } from '../../utils/connect';
+
+import './TrackCounter.css';
+
+type OwnProps = {|
+  +counterIndex: CounterIndex,
+|};
+
+type StateProps = {|
+  +counter: Counter,
+  +threadIndex: ThreadIndex,
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+|};
+
+type DispatchProps = {|
+  updatePreviewSelection: typeof updatePreviewSelection,
+|};
+
+type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
+
+type State = {||};
+
+export class TrackVisualProgressImpl extends React.PureComponent<Props, State> {
```

Comment:
Please add some documentation to the top of this class explaining what's going on with it.

Reply:
I believe this comment still needs to be handled.